src/rust/src/packed_r_tree.rs

```diff
@@ -45,6 +47,39 @@ impl NodeItem {
         }
     }
 
+    fn from_bytes(raw: &[u8]) -> Result<Self> {
+        if raw.len() < size_of::<NodeItem>() {
+            return Err(GeozeroError::GeometryIndex);
+        }
+        let conv_err = { |_| GeozeroError::GeometryIndex };
+        Ok(NodeItem {
+            min_x: f64::from_le_bytes(raw[0..8].try_into().map_err(conv_err)?),
```

Comment:
Do you think it'd be worth de-duping the offset logic with something like:

```
Self::from_reader(&mut std::io::Cursor::new(raw))
```

Reply:
Thanks for the great tip, I always forget about that!